### ISSUE TITLE
Speed up GHA workflows

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3book#760]
Because we render with `--cache-refresh` and we render to both pdf and html, all the code from the book is run twice in each workflow. This unnecessarily slows down the rendering time and we can / should do something about this.

**Comment by jemus42** [comment]
Why do we render with `--cache-refresh` on GitHub?

I know that option came up when we tried debugging people's issues with local rendering, where it made sense to occasionally throw out the cache.
On GitHub, I don't see what the benefit would be, as we explicitly cache things that need caching and any other caching mechanism outside of the GHA cache does not have an effect anyway, right?

**Comment by sebffischer** [comment]
Yeah, I also though about this, at least we should run everything with `--no-cache` instead of `--cache-refresh` if we don't use the cache anyway

**Comment by jemus42** [comment]
Wouldn't that mean that the second rendering wouldn't benefit from the first one, so the code would still run twice per workflow?

**Comment by sebffischer** [comment]
Yes, I think we should probably just enable caching. Currently we are not only not caching, we are creating a cache that we then throw away, which is like the worst thing we can do :D

**Comment by sebffischer** [comment]
Yeah, I would not cache **between** workflows, but just **within** a workflow

**Comment by be-marc** [comment]
The flag is useless unless someone accidentally uploaded their local cache. But it seems that the book is no longer rendered twice. See #890 . Maybe quarto changed this. Close?
